core/src/Kokkos_Printf.hpp

```diff
@@ -39,15 +42,13 @@ KOKKOS_FORCEINLINE_FUNCTION void printf(const char* format, Args... args) {
   else
     sycl::ext::oneapi::experimental::printf(format, args...);
 #else
-  if constexpr (sizeof...(Args) == 0) ::printf("%s", format);
-    // FIXME_OPENMPTARGET non-string-literal argument used in printf is not
-    // supported for spir64
-#if !(defined(KOKKOS_ENABLE_OPENMPTARGET) && defined(KOKKOS_ARCH_INTEL_GPU))
+  if constexpr (sizeof...(Args) == 0)
+    ::printf("%s", format);
```

Comment:
Do we still need handle no arguments passed separately with these changes?

Reply:
Yes, this still avoids warnings (I don't exactly recall which exact compiler and compiler versions) and was part of the implementation before we made changes for `OpenMPTarget`, see https://github.com/kokkos/kokkos/pull/6083/files.